### ISSUE TITLE
Attempted to set 'backend' to 'secret' on the deleted record <secret:test/test/test/test> when creating secret at same path as deleted secret

### ISSUE THREAD

**Issue by rayterrill** [hashicorp/vault#9321]
**Describe the bug**
We're seeing the error "Attempted to set 'backend' to 'secret' on the deleted record <secret:test/test/test/test>" in the GUI when we attempt to recreate a secret that was deleted. Refreshing the GUI sometimes shows that the secret is actually created.

**To Reproduce**
1. Delete a secret in the GUI that results in a path/folder being deleted.
2. Attempt to recreate the secret using the GUI
3. See error

Note that we cannot appear to recreate this error using the CLI.

**Expected behavior**
The secret should be created with no errors.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.4.2
* Vault CLI Version (retrieve with `vault version`): 1.3.4
* Server Operating System/Architecture: Ubuntu 18.04.2 LTS

Vault server configuration file(s):

```hcl
storage "mssql" {
  server = "REMOVED"
  username = "vault"
  password = "REMOVED"
  database = "Vault"
  table = "vault"
  appname = "vault"
  schema = "dbo"
  connectionTimeout = 30
  logLevel = 0
}

ui = true

listener "tcp" {
  address = "REMOVED:443"
  tls_disable = 0
  tls_cert_file = "/etc/vault.d/REMOVED.crt"
  tls_key_file  = "/etc/vault.d/REMOVED.key"
  tls_disable_client_certs = 1
}
```

**Additional context**

**Comment by msadofschi** [comment]
It happened to me now.
If you close the browser and open a new one you will see the secret is there.

**Comment by btkrausen** [comment]
This bug is still present in Vault 1.7.3+Ent

**Comment by hellobontempo** [comment]
Hi there! Thanks for reporting this 🐛  I've been unable to reproduce the issue. I'm wondering if the issue has been resolved in a newer release of Vault (>=1.11.x), or if it's still a problem could more detailed reproduction steps could be provided?

**Comment by hellobontempo** [comment]
There have been quite a few improvements to this area of the codebase, closing out as unable to reproduce
